client/src/components/UserInfoStack.tsx

```diff
@@ -0,0 +1,34 @@
+import { formatDate, Job, Application } from '../lib';
+import { Link } from 'react-router-dom';
+import Button from './Button';
+
+type InfoStackProps = {
+  jobApplication: (Application & Job)[] | undefined;
```

Comment:
Same comment here about accepting `undefined`. Ideally, you would never allow `undefined` for any props on any component. But that's not always possible or convenient. I would start with "must be defined" and add `undefined` only if things get really difficult.